### PR TITLE
feat(cut-short): Implement cut-short integration

### PR DESCRIPTION
**Pull request by Fryuni**
The new package is a more friendly alternative for:
- https://github.com/astrolicious/astro-tips.dev/issues/242

Instead of having to implement an extra middleware with the logic and have one of these in your code:
- ```ts
  throw Object.assign(redirect(login_path(url), 302), {
    __internal: true,
  });
  ```
- ```ts
  throw new ErrorWithResponse('something', () => redirect(login_path(url), 302));
  ```

You can add the integration and just do this and don't worry about writing a middleware:
```ts
endRequest(redirect(login_path(url), 302))
```

### CODE REVIEW

File: docs/src/content/docs/cut-short.mdx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+From any code that is reachable from a page rendering context can use the `endRequest` function to stop the.
+
+A page-rendering context is when you are inside of:
+
+- A middleware;
+- The frontmatter of a page component (not components _in_ the page, see [streaming](#streaming));
+- An API endpoint;
+- A function called from another page-rendering context.
+
+```ts title="src/pages/userId.ts" del={3-4} ins={7-8,18-19} {10-11}
```
Comment (OliverSpeir) [inline docs/src/content/docs/cut-short.mdx:26]: I think this example is a bit complicated, 

to simplify I'd remove the invalid example and maybe even make another file like example.ts or something where you define the getUser function and import it into the api endpoint and page.

File: packages/utils/src/types.ts [hunk 2]
```diff
@@ -2,3 +2,9 @@
 export type Prettify<T> = {
 	[K in keyof T]: T[K];
 } & {};
+
+export type MaybePromise<T> = T | Promise<T>;
+
+export type MaybeFactory<T> = T | (() => T);
+
+export type MaybeThunk<T> = MaybeFactory<MaybePromise<T>>;
```
Comment (OliverSpeir) [inline packages/utils/src/types.ts:10]: Doesn't seem like too big of a deal but I figured I'd mention, this name didn't click instantly in my head

**Review by OliverSpeir — APPROVED** [review]
LGTM 

Impressive as always how quickly you can spin up these packages complete with tests and everything
 
 This one grew on me more and more as I read through it

File: packages/utils/src/types.ts (hunk 2 above)
Comment (Fryuni) [reply packages/utils/src/types.ts:10]: Seemed like the appropriate name for me. It comes from functional programming

> 'thunk' is a function that takes no arguments and, when invoked, performs a potentially expensive operation

Any suggestions that would make this more clear? Would a JSDoc have helped?

Comment (OliverSpeir) [reply packages/utils/src/types.ts:10]: Ah I just didn't know the term, I suspected it was something meaningful I didn't know about haha

A JSDoc could be cool, but I don't think its truly necessary anyone curious would probably google the term

**Pull request merged**